lib/spacesuit/proxy_handler.ex

```diff
@@ -29,16 +29,17 @@ defmodule Spacesuit.ProxyHandler do
         stream(upstream, downstream)
 
       {:error, :econnrefused} ->
-        error_reply(req, 503, "Service Unavailable - Connection refused")
+        error_reply(req, 503, "SERVICE_UNAVAILABLE", "Connection refused")
 
       {:error, :closed} ->
-        error_reply(req, 502, "Bad Gateway - Connection closed")
+        error_reply(req, 502, "BAD_GATEWAY", "Connection closed")
 
       {:error, :timeout} ->
-        error_reply(req, 502, "Bad Gateway - Connection timeout")
+        error_reply(req, 502, "BAD_GATEWAY", "Connection timeout")
 
       {:error, :bad_request} ->
-        error_reply(req, 400, "Bad Request")
+        # What is this doing, the response for any bad request should be streamed through
+        error_reply(req, 400, "Bad Request", :nil)
```

Comment:
This only happens when something is wrong with the request and we can't even send it to the upstream host. Something like a badly formatted URL, for example. When the request is able to be sent to the upstream, Spacesuit just shuffles all that data back.